server/server.js

```diff
@@ -1047,51 +1047,78 @@ let needSetup = false;
             }
         });
 
-        socket.on("deleteMonitor", async (monitorID, callback) => {
+        socket.on("deleteMonitor", async (monitorID, deleteChildren, callback) => {
             try {
-                checkLogin(socket);
-
-                log.info("manage", `Delete Monitor: ${monitorID} User ID: ${socket.userID}`);
-
-                if (monitorID in server.monitorList) {
-                    await server.monitorList[monitorID].stop();
-                    delete server.monitorList[monitorID];
+                // Backward compatibility: if deleteChildren is omitted, the second parameter is the callback
+                if (typeof deleteChildren === "function") {
+                    callback = deleteChildren;
+                    deleteChildren = false;
                 }
```

Comment:
@BigBoot could you review how to make this not breaking for your usecase?

I would prefer it like this, but am not sure if this would break you.

```suggestion
```

Reply:
Just took a look how socketio handles this and it seems like that would indeed be needed for backwards compatibility. Given I don't have a stable release for uptime kuma v2 yet, breaking now would be fine from my side too though. In that case I would need to select the correct signature based on the uptime kuma version (i.e.  2.1.x+ ? Or would this be coming to 2.0.x too?)